src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests2.cs

```diff
@@ -0,0 +1,205 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    [CompilerTrait(CompilerFeature.Patterns)]
+    public class PatternMatchingTests2 : PatternMatchingTestBase
+    {
+        [Fact]
+        public void Patterns2_00()
+        {
+            var source =
+@"
+using System;
+class Program
+{
+    public static void Main()
+    {
+        Console.WriteLine(1 is int {} x ? x : -1);
+    }
+}
+";
+            var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugExe, parseOptions: TestOptions.RegularWithRecursivePatterns);
```

Comment:
Why `CreateCompilationWithMscorlib45`? Is there something 4.5 specific about these tests?

Reply:
I don't think so. I suspect I was just copying some old tests.

---
In reply to: [155100247](https://github.com/dotnet/roslyn/pull/23209#discussion_r155100247) [](ancestors = 155100247)

Reply:
Recorded in [https://github.com/dotnet/roslyn/issues/23222](https://github.com/dotnet/roslyn/issues/23222)

---
In reply to: [155120874](https://github.com/dotnet/roslyn/pull/23209#discussion_r155120874) [](ancestors = 155120874,155100247)